sdk/storage/storage-queue/src/QueueClient.ts

```diff
@@ -538,6 +743,254 @@ export class QueueClient extends StorageClient {
+   * @memberof QueueClient
+   */
+  public async clearMessages(
+    options: MessagesClearOptions = {}
+  ): Promise<Models.MessagesClearResponse> {
+    const { span, spanOptions } = createSpan("QueueClient-clearMessages", options.spanOptions);
+    try {
+      return this.messagesContext.clear({
+        abortSignal: options.abortSignal,
+        spanOptions
+      });
+    } catch (e) {
+      span.setStatus({
+        code: CanonicalCode.UNKNOWN,
+        message: e.message
+      });
+      throw e;
+    } finally {
+      span.end();
+    }
+  }
+
+  /**
+   * Enqueue adds a new message to the back of a queue. The visibility timeout specifies how long
+   * the message should be invisible to Dequeue and Peek operations.
+   * The message content is up to 64KB in size, and must be in a format that can be included in an XML request with UTF-8 encoding.
+   * To include markup in the message, the contents of the message must either be XML-escaped or Base64-encode.
+   * @see https://docs.microsoft.com/en-us/rest/api/storageservices/put-message
+   *
+   * @param {string} messageText Text of the message to enqueue
+   * @param {MessagesEnqueueOptionas} [options] Options to Messages enqueue operation.
+   * @returns {Promise<Models.MessagesEnqueueResponse>} Response data for the Messages enqueue operation.
+   * @memberof QueueClient
+   */
+  public async enqueueMessage(
+    messageText: string,
+    options: MessagesEnqueueOptions = {}
+  ): Promise<MessagesEnqueueResponse> {
+    const { span, spanOptions } = createSpan("QueueClient-enqueueMessage", options.spanOptions);
+    try {
+      const response = await this.messagesContext.enqueue(
+        {
+          messageText: messageText
+        },
+        {
+          abortSignal: options.abortSignal,
+          ...options,
+          spanOptions
+        }
+      );
+      const item = response[0];
+      return {
+        _response: response._response,
+        date: response.date,
+        requestId: response.requestId,
+        clientRequestId: response.clientRequestId,
+        version: response.version,
+        errorCode: response.errorCode,
+        messageId: item.messageId,
+        popReceipt: item.popReceipt,
+        timeNextVisible: item.timeNextVisible,
+        insertionTime: item.insertionTime,
+        expirationTime: item.expirationTime
+      };
+    } catch (e) {
+      span.setStatus({
+        code: CanonicalCode.UNKNOWN,
+        message: e.message
+      });
+      throw e;
+    } finally {
+      span.end();
+    }
+  }
+
+  /**
+   * Dequeue retrieves one or more messages from the front of the queue.
+   * @see https://docs.microsoft.com/en-us/rest/api/storageservices/get-messages
+   *
+   * @param {MessagesDequeueOptionals} [options] Options to Messages dequeue operation.
+   * @returns {Promise<Models.MessagesDequeueResponse>} Response data for the Messages dequeue operation.
+   * @memberof QueueClient
+   */
+  public async dequeueMessages(
```

Comment:
per the latest consistency meeting. method name should be sendMessage and receiveMessages.  We can rename the result types in another PR (involving autorest).

Reply:
Done